### ISSUE TITLE
add password for accessing panel

### ISSUE THREAD

**Issue by naufalprojects** [devlikeapro/whatsapp-http-api#18]
Can you please add a password for accessing panel? So it prevent "unauthorized access"

**Comment by allburov** [comment]
@naufalprojects hi!
I don't think that panel must be protected. It's just a list of API endpoints, no secrets here.
I think **we must protect the endpoints**, like I was planning in https://github.com/allburov/whatsapp-http-api/issues/4
It'll look like you can access the panel with no password, but you can not perform any api calls without the right API key.

The best way to add a feature to the project - create a pull request with that changes, **PRs are welcome here!**

 I'm using the project inside the private network, for me the feature does not have a high priority for me.
But you you can contact me at Telegram https://t.me/allburov to discuss a feature price or support project on monthly base https://github.com/allburov/whatsapp-http-api#support

**Comment by allburov** [comment]
API Token Key authentication is available in **WAHA Plus** version https://waha.devlike.pro/docs/how-to/security/

For now you can protect API with it, later we're going to add password key on panel too

**Comment by sfnemis** [comment]
Unfortunately, API its not working :( I purchased WAHA PLUS I got an error 
![SS-Swagger UI-Google Chrome-19-11 at 03](https://user-images.githubusercontent.com/78266266/202830479-27b6a31f-0713-4cf5-94e6-5a1432882a28.png)
and also;

1.  with even with VENOM Status changes online / offline is not working it shows always false .. :( 
2. WHATSAPP_ START SESSION=example it's not working 
I upload also my ENV Settings you can see 
![SS-Portainer  local-Google Chrome-19-11 at 05](https://user-images.githubusercontent.com/78266266/202830938-a0f118ff-692e-4c10-923d-8b4a4b126e25.png)

Thank you @allburov

**Comment by allburov** [comment]
@sfnemis thank you for providing such detail feedback! It's really helpful, appreciate it! 👏

0. API Key authentication fixed in WAHA Plus `2022.11.19` (the `latest` version too) - **please pull the new version image and try again!**
1. I'm not sure what does it mean - do you mean `state.change` event send always false? Could you share the payload of the webhook please?
2. I've tested WHATSAPP_START_SESSION with  the below command - it works! Have you tried it with plus version? Because core version support only `default` session - you either should set `WHATSAPP_START_SESSION=default` or run Plus version 
```bash
docker run --rm -p 127.0.0.1:3000:3000/tcp -e WHATSAPP_DEFAULT_ENGINE=VENOM -e WHATSAPP_START_SESSION=wazzup --name whatsapp-http-api devlikeapro/whatsapp-http-api-plus
```

Could you please create separate issues for 1\2 problems and give more information if possible - logs are helpful too, you can get them by running `docker logs whatsapp-http-api` 


Thank you for supporting the project, I hope the project will help you to achieve your goals! 🚀

**Comment by allburov** [comment]
Hi!
Here's the command that you can use to hide and protect Swagger panel with `admin/admin` credentials
```
docker run -it -e WHATSAPP_SWAGGER_USERNAME=admin -eWHATSAPP_SWAGGER_PASSWORD=admin devlikeapro/whatsapp-http-api-plus
```

It enables basic authentication http://localhost:3000 page, like this:
![swagger-basic-auth](https://user-images.githubusercontent.com/4376814/206897310-86879039-37c4-4a7e-bd4b-61be9683632c.png)

Available in WAHA Plus `2022.12.11` and higher, please pull the new version before testing!
